### PR TITLE
fix: make sure _assessFees isn't called twice within the same block

### LINKED ISSUE #322: Management fee can be avoided
Block `report` from being called if time since last report is 0 (same block)

### PR DESCRIPTION
**Pull request by steffenix**
fixes #322

### CODE REVIEW

File: tests/functional/strategy/test_migration.py [hunk 1]
```diff
@@ -4,9 +4,11 @@
 
 
 def test_good_migration(
-    token, strategy, vault, gov, strategist, guardian, TestStrategy, rando
+    token, strategy, vault, gov, strategist, guardian, TestStrategy, rando, chain
 ):
     # Call this once to seed the strategy with debt
+    chain.sleep(1)
+    chain.sleep(1)
```
Comment (fubuloubu) [inline tests/functional/strategy/test_migration.py:11]: Why is this here twice?

**Review by fubuloubu — APPROVED** [review]

**Comment by zgfzgf** [comment]
assert duration != 0 should be in report function

**Comment by steffenix** [comment]
> assert duration != 0 should be in report function

It's more gas efficient to have it within `_assessFees` for strategies that do not misbehave.

**Comment by zgfzgf** [comment]
I see. thank you
